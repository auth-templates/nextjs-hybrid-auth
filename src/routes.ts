export const PrivateRoutes = {
    dashboard: '/dashboard',
    account: '/account',
    leaves: '/leaves'
};

export const PublicRoutes = {
    login: '/login',
    register: '/register',
    recoverPassword: '/recover-password',
    resendConfirmationEmail: '/resend-confirmation-email',
    resetPassword: '/reset-password/:token',
    verifyAccount: '/verify-account/:token',
    twoStep: '/two-step',
}