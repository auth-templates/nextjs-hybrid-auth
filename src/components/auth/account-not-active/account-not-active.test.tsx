import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import AccountNotActive from './account-not-active';
import { PublicRoutes } from '../../../routes';

describe("AccountNotActive", () => {
    it('should contain all elements', async () => {
        render(
            <MemoryRouter>
                <AccountNotActive email={'account@mail.com'} />
            </MemoryRouter>
        );

        expect(screen.getByText(/An account with the email/)).toBeInTheDocument(); 
        expect(screen.getByText(/account@mail.com/)).toBeInTheDocument();
        expect(screen.getByText(/exists, but it is not active./)).toBeInTheDocument();
        expect(screen.getByText(/Check your inbox for the confirmation email or please go to/)).toBeInTheDocument();
        expect(screen.getByRole('link', { name: 'Resend confirmation email' })).toHaveAttribute('href', PublicRoutes.resendConfirmationEmail);
        expect(screen.getByText(/page in order to send a new confirmation email./)).toBeInTheDocument();
    });

    it("renders correctly", () => {
        const tree = renderer.create(
            <MemoryRouter>
                <AccountNotActive email={'account@mail.com'} /> 
            </MemoryRouter>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});