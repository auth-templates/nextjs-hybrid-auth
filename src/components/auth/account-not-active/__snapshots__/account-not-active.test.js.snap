// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountNotActive renders correctly 1`] = `
<div
  className="accountNotActive"
>
  An account with the email 
  <div
    className="email"
  >
    account@mail.com
  </div>
   exists, but it is not active. Check your inbox for the confirmation email or please go to
  <a
    href="/resend-confirmation-email"
    onClick={[Function]}
  >
     Resend confirmation email 
  </a>
  page in order to send a new confirmation email.
</div>
`;
